test/strings/search.jl

```diff
@@ -29,7 +29,7 @@ end
 # @test_throws ErrorException in("foobar","bar")
 @test_throws BoundsError search(b"\x1\x2",0x1,0)
 @test rsearchindex(b"foo",b"o",0) == 0
-@test rsearchindex(SubString("",1,1),SubString("",1,1)) == 1
+@test_throws BoundsError rsearchindex(SubString("",1,1),SubString("",1,1)) # == 1
```

Comment:
Should return `1` just like for `String`, right?

Reply:
Should throw `BoundsError` as `SubString("",1,1)` throws such error. I have added another test case which should return `1`.

Reply:
Ah, yes, of course! But then it doesn't really make sense to pretend calling `rsearchindex` when the exception is thrown by the `SubString` constructor. Let's just remove that line and only keep the one that succeeds.

Reply:
Done